src/iosrv/module.c

```diff
@@ -179,18 +201,20 @@ dss_module_load(const char *modname, uint64_t *mod_facs)
 	dss_unregister_key(smod->sm_key);
 	smod->sm_fini();
 err_lmod:
+	d_list_del_init(&lmod->lm_lk);
```

Comment:
Do you need lock here?

Reply:
this is called from module_init_all which holds the lock